device/src/main/java/com/softserve/lv460/device/config/ChangeStreamConfig.java

```diff
@@ -0,0 +1,123 @@
+
+import javax.annotation.PostConstruct;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+@Service
+public class ChangeStreamConfig {
+  private final MongoClient mongoClient;
+  private final CloseableHttpClient httpClient;
+  private final PropertiesConfig propertiesConfig;
+  private final ActionExecutor actionExecutor;
+  private LoadingCache<String, List<RuleDto>> cache;
+  private static Logger logger = LoggerFactory.getLogger(DeviceDataController.class);
+
+  public ChangeStreamConfig(MongoClient mongoClient, CloseableHttpClient httpClient, PropertiesConfig propertiesConfig, ActionExecutor actionExecutor) {
+    this.httpClient = httpClient;
+    this.propertiesConfig = propertiesConfig;
+    this.actionExecutor = actionExecutor;
+    this.mongoClient = mongoClient;
+    this.cache = CacheBuilder
+            .newBuilder()
+            .expireAfterWrite(propertiesConfig.getCacheExpiration(), TimeUnit.MINUTES)
+            .build(
+                    new CacheLoader<String, List<RuleDto>>() {
+                      @Override
+                      public List<RuleDto> load(String uuId) throws IOException {
+                        return getRules(uuId);
+                      }
+                    }
+            );
+  }
+
+  @PostConstruct
+  public void subscribeToChangeStream() {
+    Thread thread = new Thread(() -> {
+      MongoDatabase db = mongoClient.getDatabase(propertiesConfig.getDatabase());
+      MongoCollection<Document> collection = db.getCollection(propertiesConfig.getCollection());
+      Block<ChangeStreamDocument<Document>> printBlock = changeStreamDocument -> {
+        Document fullDocument = changeStreamDocument.getFullDocument();
+        List<RuleDto> rules = getCache(fullDocument.getString("uuId"));
+        for (RuleDto rule : rules) {
+          if (checkRuleCondition(rule, (Document) fullDocument.get("data"))) {
+            for (ActionRule actionRule : rule.getActionRule()) {
+              actionExecutor.doAction(actionRule);
+            }
+          }
+        }
+      };
+      collection.watch().forEach(printBlock);
+    });
+    thread.start();
+  }
+
+  private List<RuleDto> getCache(String uuId) {
+    try {
+      return cache.get(uuId);
+    } catch (Exception e) {
+      logger.error(e.getLocalizedMessage());
+      return Collections.emptyList();
+    }
+  }
+
+  private List<RuleDto> getRules(String uuId) throws IOException {
+    String deviceRuleUrl = propertiesConfig.getMainApplicationHostName() + "/rules/device/" + uuId;
+    CloseableHttpResponse response = httpClient.execute(new HttpGet(deviceRuleUrl));
+    return new ObjectMapper().readValue(response.getEntity().getContent(),
+            new TypeReference<List<RuleDto>>() {
+            });
+  }
+
+  private Boolean checkRuleCondition(RuleDto rule, Document data) {
+    try {
+      JSONObject conditionJson = new JSONObject(rule.getConditions());
```

Comment:
You've added support only for Integers. What about decimals, enums? we agreed that we support those types as well.